src/test/java/test/Test1.java

```diff
@@ -36,7 +38,8 @@ public void includedGroups() {
 
         tng.run();
 
-        assertThat(listener.getSucceedMethodNames()).containsExactly("method1", "method3");
+        List<String> expected = Arrays.asList("method1", "method3");
+        doAssert(listener.getSucceedMethodNames(), expected);
```

Comment:
All the tests that have been modified are only making themselves agnostic to the order of the test methods. The current tests have all had this hard wired into them. The moment I removed off the ordering, the tests started breaking because now the order is all governed by how Java's reflection yields those methods.

> An option could be to introduce a new edge between tests of the same class.

Where would that new option reside ? It would be good if you can please help add more context.